### DISCUSSION TITLE
Video Streaming or lack there of

### DISCUSSION THREAD

**Discussion by connectedway** [moneytoo/Player#457] [General]
Does Player have a streaming mode.  Particularly when content is off device and using Storage Access Framework to access?  Player works great for a small video, say a music video.  But it performs poorly with a large video file.  Observing network traffic, it appears to read the entire file, no matter how big, twice before starting to play.  Then it appears to not adequately buffer the content which ends up resulting in lags during play.

**Comment by moneytoo** [comment]
When a file is opened, FFmpeg/ffprobe sniffs it to detect chapters and frame rate. This shouldn't generally make the file read in full. However most document providers do exactly that - they download the files in full as they cannot seek.
